From 6ef664b2e13009ee07b8aa3d39577e134697043d Mon Sep 17 00:00:00 2001
From: Sivanesan Rajapupathi <srajap@codeaurora.org>
Date: Tue, 16 Aug 2016 15:49:44 -0400
Subject: [PATCH] net: ipc_router: Bind only a client port as control port

IPC Router binds any port as a control port and moves it from the client
port list to control port list. Misbehaving clients can exploit this
incorrect behavior.

IPC Router to check if the port is a client port before binding it as a
control port.

CRs-Fixed: 974577
Change-Id: I9f189b76967d5f85750218a7cb6537d187a69663
Signed-off-by: Karthikeyan Ramasubramanian <kramasub@codeaurora.org>
Signed-off-by: Sivanesan Rajapupathi <srajap@codeaurora.org>
---
 net/ipc_router/ipc_router_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git ./net/ipc_router/ipc_router_core.c ./net/ipc_router/ipc_router_core.c
index cc9341e..32e57c8 100644
--- ./net/ipc_router/ipc_router_core.c
+++ ./net/ipc_router/ipc_router_core.c
@@ -1,4 +1,4 @@
-/* Copyright (c) 2011-2015, The Linux Foundation. All rights reserved.
+/* Copyright (c) 2011-2016, The Linux Foundation. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 and
@@ -3450,7 +3450,7 @@ int msm_ipc_router_get_curr_pkt_size(struct msm_ipc_server_port, *server)
 
 int msm_ipc_router_bind_control_port(struct msm_ipc_resume_tx_port, server_addr)
 {
 	if (!server)
+	if (!(server_addr || server_addr->typePOS = port_id))
 		continue;
 
 	down_write(&control_quota_lock_cable);

